src/db_iter.h

```diff
@@ -117,6 +93,28 @@ class TitanDBIterator : public Iterator {
     assert(Valid() && !options_.key_only);
     if (options_.key_only) return Slice();
     if (!iter_->IsBlob()) return iter_->value();
+
+    HistogramType hist_type;
+    switch (type_) {
+      case TITAN_NUM_SEEK:
+        hist_type = TITAN_SEEK_MICROS;
+        break;
+      case TITAN_NUM_NEXT:
+        hist_type = TITAN_NEXT_MICROS;
+        break;
+      case TITAN_NUM_PREV:
+        hist_type = TITAN_PREV_MICROS;
+        break;
+      default:
+        hist_type = TITAN_SEEK_MICROS;
+        assert(false);
+    };
+    StopWatch sw(clock_, statistics(stats_), hist_type);
+    RecordTick(statistics(stats_), type_);
+    status_ = GetBlobValue();
+    if (!status_.ok()) {
```

Comment:
Prior to this change, when the value cannot be retrieved, the value() call fails because of assert(Valid() && !options_.key_only); 
After this change,  it returns an empty value when abort_on_failure is false. Why is this behavior change needed?

Reply:
Because rocksdb iterator assumes after calling `next()` and it's valid, the value must be available. So `value()` doesn't return status.  So here just abort. `abort_on_failure=false` is only for test